cmd/thanos/query.go

```diff
@@ -500,6 +510,20 @@ func runQuery(
 					specs = append(specs, tmpSpecs...)
 				}
 
+				for _, eg := range endpointGroupAddrs {
+					addr := fmt.Sprintf("dns:///%s", eg)
+					spec := query.NewGRPCEndpointSpec(addr, false, extgrpc.EndpointGroupGRPCOpts()...)
+					specs = append(specs, spec)
+				}
+
+				for _, eg := range strictEndpointGroups {
+					grpc.WithDisableRetry()
+
+					addr := fmt.Sprintf("dns:///%s", eg)
+					spec := query.NewGRPCEndpointSpec(addr, true, extgrpc.EndpointGroupGRPCOpts()...)
```

Comment:
Should `grpc.WithDisableRetry` be passed along side the other grpc options?
```suggestion
					addr := fmt.Sprintf("dns:///%s", eg)
					spec := query.NewGRPCEndpointSpec(addr, true, append(extgrpc.EndpointGroupGRPCOpts(), grpc.WithDisableRetry())...)
```

Reply:
Ah, I think this like should be completely removed. The retry should always be there as explained in the comment below.